### ISSUE TITLE
`regularized_incomplete_beta_p` incorrectly broadcasts when `vmap`-ed

### ISSUE THREAD

**Issue by sharadmv** [google/jax#3813]
Repro:
```python
from functools import partial
from jax import lax
import jax.numpy as jnp
import jax

reg_inc_beta = partial(lax.regularized_incomplete_beta_p.bind, jnp.ones(1), 0.5)
print(reg_inc_beta(jnp.array(1.))) # works
print(jax.vmap(reg_inc_beta)(jnp.ones([4]))) # ERRORS
```

Traceback:
```
[1.]
/home/sharadmv/workspace/jax/jax/lax/lax.py:1758: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  shapes = onp.array([aval.shape for aval in avals if aval.shape])
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    print(jax.vmap(reg_inc_beta)(jnp.ones([4]))) # ERRORS
  File "/home/sharadmv/workspace/jax/jax/api.py", line 770, in batched_fun
    lambda: flatten_axes(out_tree(), out_axes))
  File "/home/sharadmv/workspace/jax/jax/interpreters/batching.py", line 34, in batch
    return batched_fun.call_wrapped(*in_vals)
  File "/home/sharadmv/workspace/jax/jax/linear_util.py", line 150, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/home/sharadmv/workspace/jax/jax/core.py", line 214, in bind
    out_tracer = top_trace.process_primitive(self, tracers, kwargs)
  File "/home/sharadmv/workspace/jax/jax/interpreters/batching.py", line 134, in process_primitive
    val_out, dim_out = batched_primitive(vals_in, dims_in, **params)
  File "/home/sharadmv/workspace/jax/jax/interpreters/batching.py", line 249, in broadcast_batcher
    out = prim.bind(*args, **params)
  File "/home/sharadmv/workspace/jax/jax/core.py", line 211, in bind
    return self.impl(*args, **kwargs)
  File "/home/sharadmv/workspace/jax/jax/interpreters/xla.py", line 217, in apply_primitive
    compiled_fun = xla_primitive_callable(prim, *map(arg_spec, args), **params)
  File "/home/sharadmv/workspace/jax/jax/interpreters/xla.py", line 231, in xla_primitive_callable
    aval_out = prim.abstract_eval(*avals, **params)
  File "/home/sharadmv/workspace/jax/jax/lax/lax.py", line 1712, in standard_abstract_eval
    return ShapedArray(shape_rule(*args, **kwargs), dtype_rule(*args, **kwargs))
  File "/home/sharadmv/workspace/jax/jax/lax/lax.py", line 1763, in _broadcasting_shape_rule
    raise TypeError(msg.format(name, ', '.join(map(str, map(tuple, shapes)))))
TypeError: regularized_incomplete_beta got arrays of different rank: (4, 1), (4,), (4, 1).
```

**Comment by jakevdp** [comment]
Thanks - looks like this will need a custom batching rule.

**Comment by froystig** [comment]
fixed by #6145
